includes/Transport/HttpTransport.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * MCP HTTP Transport for WordPress - MCP 2025-06-18 Compliant
+ *
+ * This transport implements the MCP Streamable HTTP specification and can work
+ * both with and without the mcp-wordpress-remote proxy.
+ *
+ * @package McpAdapter
+ */
+
+declare( strict_types=1 );
+
+namespace WP\MCP\Transport;
+
+use WP\MCP\Transport\Contracts\McpRestTransportInterface;
+use WP\MCP\Transport\Infrastructure\HttpRequestContext;
+use WP\MCP\Transport\Infrastructure\HttpRequestHandler;
+use WP\MCP\Transport\Infrastructure\McpTransportContext;
+use WP\MCP\Transport\Infrastructure\McpTransportHelperTrait;
+
+/**
+ * MCP HTTP Transport - Unified transport for both proxy and direct clients
+ *
+ * Implements MCP 2025-06-18 Streamable HTTP specification
+ */
+class HttpTransport implements McpRestTransportInterface {
+	use McpTransportHelperTrait;
+
+	/**
+	 * The HTTP request handler.
+	 *
+	 * @var \WP\MCP\Transport\Infrastructure\HttpRequestHandler
+	 */
+	protected HttpRequestHandler $request_handler;
+
+	/**
+	 * Initialize the class and register routes
+	 *
+	 * @param \WP\MCP\Transport\Infrastructure\McpTransportContext $transport_context The transport context.
+	 */
+	public function __construct( McpTransportContext $transport_context ) {
+		$this->request_handler = new HttpRequestHandler( $transport_context );
+		add_action( 'rest_api_init', array( $this, 'register_routes' ), 16 );
```

Comment:
Can we move this to an `initialize()` method or somewhere else? I think it's better to not have hooks as a side-effect of constructors.